src/profile-logic/import/instruments/index.js

```diff
@@ -0,0 +1,910 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import {
+  getEmptyProfile,
+  getEmptyThread,
+} from '../../../profile-logic/data-structures';
+import BinaryPlistParser, { UID } from './BinaryPlistParser';
+import BinReader from './BinReader';
+import MaybeCompressedReader from './MaybeCompressedReader';
+
+//types
+import type {
+  Profile,
+  FuncTable,
+  FrameTable,
+  StackTable,
+} from '../../../types/profile';
+import type { UniqueStringArray } from '../../../../src/utils/unique-string-array';
+
+type InstrumentsSample = {|
+  +timestamp: number,
+  +threadID: number,
+  +backtraceID: number,
+  backtraceStack?: Array<number>,
```

Comment:
I think it would be a good idea to have 2 different types: one with `backtraceStack` and one without. So that you don't need to check for its existence when you want to use it.

Reply:
@julienw, I have used `backtraceIDtoStack` separately. So now we don't need two types here. I have pushed the changes. Thanks! :)